compiler-core/src/ast/typed.rs

```diff
@@ -152,7 +152,12 @@ pub enum TypedExpr {
         args: Vec<TypedRecordUpdateArg>,
     },
 
-    Negate {
+    NegateBool {
+        location: SrcSpan,
+        value: Box<Self>,
+    },
+
+    NegateInteger {
```

Comment:
```suggestion
    NegateInt {
```

To be consistent within the codebase 🙏